doc/SONiC_OC_NTP_HLD.md

```diff
@@ -0,0 +1,491 @@
+      |  |     |  +--rw key-id?      uint16
+      |  |     |  +--rw key-type?    identityref
+      |  |     |  +--rw key-value?   string
+      |  |     +--ro state
+      |  |        +--ro key-id?      uint16
+      |  |        +--ro key-type?    identityref
+      |  |        +--ro key-value?   string
++     |  +--rw servers
++     |     +--rw server* [address]
++     |        +--rw address    -> ../config/address
++     |        +--rw config
++     |        |  +--rw address?            oc-inet:host
+      |        |  +--rw port?               oc-inet:port-number
+      |        |  +--rw version?            uint8
+      |        |  +--rw association-type?   enumeration
+      |        |  +--rw iburst?             boolean
+      |        |  +--rw prefer?             boolean
++     |        +--ro state
++     |           +--ro address?              oc-inet:host
+      |           +--ro port?                 oc-inet:port-number
+      |           +--ro version?              uint8
+      |           +--ro association-type?     enumeration
+      |           +--ro iburst?               boolean
+      |           +--ro prefer?               boolean
++     |           +--ro stratum?              uint8
++     |           +--ro root-delay?           uint32
++     |           +--ro root-dispersion?      uint64
++     |           +--ro offset?               uint64
++     |           +--ro poll-interval?        uint32
++     |           +--ro oc-sys-ext:selMode?   string
++     |           +--ro oc-sys-ext:refid?     inet:host
++     |           +--ro oc-sys-ext:type?      string
++     |           +--ro oc-sys-ext:now?       uint32
++     |           +--ro oc-sys-ext:reach?     uint8
+
+
+module: sonic-system-ntp
+
++    +--rw sonic-system-ntp
++       +--rw NTP
++       |  +--rw NTP_LIST* [global_key]
++       |     +--rw global_key    enumeration
++       |     +--rw source?       union
++       |     +--rw vrf?          string
++       +--rw NTP_SERVER
++          +--rw NTP_SERVER_LIST* [server_address]
++             +--rw server_address    inet:host
+
+```
+
+### 3.6.2 CLI
+
+
+#### 3.6.2.1 Configuration Commands
+All commands are executed in `configuration-view`:
+```
+sonic# configure terminal
+sonic(config)#
+```
+
+##### 3.6.2.1.1 Configure NTP server
+```
+sonic(config)#ntp
+    server              Configure NTP server
+sonic(config)#ntp server 
+String  NTP server address or name
+
+sonic(config)# ntp server 10.11.0.1
+
+sonic(config)# ntp server 2001:aa:aa::a
+
+sonic(config)# ntp server pool.ntp.org
+
+```
+
+##### 3.6.2.1.2 Delete NTP server
+
+```
+sonic(config)# no ntp server
+  String  NTP server address or name
+
+sonic(config)# no ntp server 10.11.0.1
+
+sonic(config)# no ntp server 2001:aa:aa::a
+
+sonic(config)# no ntp server pool.ntp.org
+
+```
+
+##### 3.6.2.1.3 Configure NTP source ip
+
+```
+sonic(config)# ntp
+    source              Configure source IP address
+
+sonic(config)# ntp source 11.22.33.55

```

Comment:
Since multiple servers are supported, how do we associate source-ip with a server.
Do we need to extend the command ntp server <address> source-ip <address>

Reply:
Source ip configuration is at the NTP global level, not per NTP server.